darwin/dataset/identifier.py

```diff
@@ -8,8 +9,11 @@ def __init__(self, dataset_slug: str, team_slug: Optional[str] = None, version:
         self.version = version
 
     @classmethod
-    def parse(cls, identifier: str):
-        team_slug, dataset_slug, version = parse(identifier)
+    def parse(cls, identifier: Union[str, "DatasetIdentifier"]) -> "DatasetIdentifier":
```

Comment:
Can't we use:

```
from __future__ import annotations

def parse(cls, identifier: Union[str, "DatasetIdentifier"]) -> DatasetIdentifier:
```

instead?

Reply:
I think it's not going to work for Python 3.6, but let me try that

Reply:
As I anticipated 😢 
https://github.com/v7labs/darwin-py/pull/167/checks?check_run_id=3086770021